### PR TITLE
Change how we configure Redis

### PR DESCRIPTION
**Pull request by malcolmbaig**
## Context

For faster test runs, we want to be able to run our test suite in parallel using multiple cores. This is some of the configuration change required to support that.

## Changes proposed in this pull request

The commits are a bit wordy but to summarize:

- We plan to use parallel_tests to split up our test suite and run each set of tests in a different process.
- Each process needs a unique Redis database to ensure test runs remain free of side effects.
- To support this, update Redis config so that it follows these conventions:
  - Use REDIS_URL wherever it's already set, meaning these changes should have no effect on any existing environment that sets this variable.
  - Use Redis database 0 in development (was previously 1, as defined by .env)
  - Use Redis database 9 when running tests normally (no change here)
  - Use Redis databases 1-n when running tests in multiple processes, where n is determined by how many processes are created.
- To ensure these conventions hold, access Redis via the `ApplyRedisConnection` class.

## Guidance to review
- Per commit recommended, commit messages give a bit more detail explaining the changes.
- Do the new conventions make sense?

## Link to Trello card
https://trello.com/c/ySrLogAF

## Things to check

- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] API release notes have been updated if necessary
- [ ] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/lib/apply_redis_connection.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+class ApplyRedisConnection
```
Comment (tijmenb) [inline app/lib/apply_redis_connection.rb:1]: Have you considered setting `Redis.current` in an initializer? If that works it would save some extra config, and new Redis-dependent dependencies (like Sidekiq) would automatically be able to fetch the connection.

**Review by tijmenb — COMMENTED** [review]
Looking good! Question about `Redis.current`

File: app/lib/apply_redis_connection.rb (hunk 1 above)
Comment (malcolmbaig) [reply app/lib/apply_redis_connection.rb:1]: Good idea! Something like this? https://github.com/DFE-Digital/apply-for-teacher-training/pull/3818/commits/b1458eb28fddd28167bac822a2d538118aa9e293

**Comment by duncanjbrown** [comment]
This looks good to me and I like the `Redis.current` approach. Does it work with `parallel_tests`?

**Comment by malcolmbaig** [comment]
> Does it work with `parallel_tests`?

Yes, just about to raise the PR adding that in.

**Review by duncanjbrown — APPROVED** [review]

**Review by tijmenb — APPROVED** [review]
LGTM!

**Review by davidgisbey — APPROVED** [review]
